httpclientservice/httpClientService.js

```diff
@@ -0,0 +1,60 @@
+import axios from 'axios';
+
+class ClientService {
+  constructor() {
+    let service = axios.create({
+      headers: {}
+    });
+    service.interceptors.response.use(this.handleSuccess, this.handleError);
+    this.service = service;
+  }
+
+  handleSuccess(response) {
+    return response;
+  }
+
+  handleError = (error) => {
+    switch (error.response.status) {
+      case 401:
+        this.redirectTo(document, '/')
+        break;
+      case 404:
+        this.redirectTo(document, '/404')
+        break;
+      default:
+        this.redirectTo(document, '/500')
+        break;
+    }
+    return Promise.reject(error)
+  }
+
+  redirectTo = (document, path) => {
+    document.location = path
+  }
+  
+  get(path, callback) {
```

Comment:
We just need the get method for the moment and the path should have a base url which in our API is common to any request and the string we want to append to that base url to complete the query.

Reply:
Let other methods to be implemented when needed. YAGNI pirnciple!!